tests/utils.py

```diff
@@ -74,3 +77,111 @@ def wait_for_server(host, port, timeout=10):
     raise TimeoutError
 
 
+
+class TestServerProcess():
+  """
+  <Purpose>
+    Creates a child process with the subprocess.Popen object and
+    TempFile object used for logging.
+
+   <Arguments>
+      log:
+        Logger which will be used for logging.
+
+      server:
+        Path to the server which will be runned by the subprocess.
+        Default is "simpler_server.py".
+
+      port:
+        The port used to access the server. If none is provided,
+        then one will be generated.
+        Default is None.
+
+      timeout:
+        Time in seconds in which the server should start or otherwise
+        TimeoutError error will be raised.
+        If 0 is given, no check if the server has started will be done.
+        Default is 10.
+
+      popen_cwd:
+        Current working directory used when instancing a
+        subprocess.Popen object.
+        Default is "."
+
+      extra_cmd_args:
+        List of additional arguments for the command
+        which will be ran from the subprocess.
+        More precisely "python -u <path_to_server> <port> <extra_cmd_args>".
+        Default is empty list.
+  """
+
+  # Starts subprocess which runs the server,
+  # sets up the TempFile, waits for server response.
+  def __init__(self, log, server='simple_server.py',
+      port=None, timeout=10, popen_cwd=".",
+      extra_cmd_args=[]):
+    # Create temporary log file used for logging stdout and stderr
+    # of the subprocess. In the mode "r+"" stands for reading and writing
+    # and "t" stands for text mode.
+    self.__temp_log_file = tempfile.TemporaryFile(mode='r+t')
+
+    self.server = server
+    self.port = port or random.randint(30000, 45000)
+    self.__logger = log
+
+    # The "-u" option forces stdin, stdout and stderr to be unbuffered.
+    command = ['python', '-u', server, str(self.port)] + extra_cmd_args
+
+    # We are reusing one server subprocess in multiple unit tests, but we are
+    # collecting the logs per test.
+    self.__server_process = subprocess.Popen(command,
+        stdout=self.__temp_log_file, stderr=subprocess.STDOUT, cwd=popen_cwd)
+
+    self.__logger.info('Server process with process id ' \
+        + str(self.__server_process.pid) + " serving on port " \
+        + str(self.port) + ' started.')
+
+    if timeout > 0:
+      try:
+        wait_for_server('localhost', self.port, timeout)
+      except Exception as e:
+        self.__logger.error(
+            "Error while waiting for the server to start: " + repr(e))
+        # Make sure that errors from the server side will be logged.
+        self.flush_log()
+        raise e
+
+
+
+  # Logs contents from TempFile, truncates buffer
+  def flush_log(self):
```

Comment:
Nit: here, and throughout `TestServerProcess`, I think this kind of comment might be more idiomatic as a docstring.
```suggestion
  def flush_log(self):
    """Logs contents from TempFile, truncates buffer"""
```